graphql_compiler/schema_generation/schema_graph.py

```diff
@@ -165,34 +136,105 @@ def class_fields(self):
     @property
     def is_vertex(self):
         """Return True if the schema element represents a vertex type, and False otherwise."""
-        return self._kind == SchemaElement.ELEMENT_KIND_VERTEX
+        return isinstance(self, Vertex)
 
     @property
     def is_edge(self):
         """Return True if the schema element represents an edge type, and False otherwise."""
-        return self._kind == SchemaElement.ELEMENT_KIND_EDGE
+        return isinstance(self, Edge)
 
     @property
     def is_non_graph(self):
         """Return True if the schema element represents a non-graph type, and False otherwise."""
-        return self._kind == SchemaElement.ELEMENT_KIND_NON_GRAPH
+        return isinstance(self, NonGraphElement)
+
+    @abstractmethod
+    def __str__(self):
```

Comment:
Not sure why the linter didn't complain here, but these functions need docstrings. Also, look at the implementation of `CompilerEntity` for an example of how you can implement `__str__` and `__repr__` generically. You'll have to do a bit of magic with the constructor, and to get the names of args defined by subtypes' constructors you'll have to pass them in as kwargs to this constructor, but it can work and simplify your code overall.

Reply:
I think that linter was in arc lint, not sure if we picked it up.